wp/lib/bap_wp/src/environment.ml

```diff
@@ -523,3 +448,9 @@ let get_init_var (env : t) (var : Var.t) : Constr.z3_expr option =
 
 let map_sub_name (env : t) (name_mod : string) : string =
   get_mapped_name name_mod env.func_name_map
+
+let get_unroll_depth (env : t) : unroll_depth =
+  env.unroll_depth
+
+let set_unroll_depth (env : t) (depth : unroll_depth) : t =
```

Comment:
do we want to wrap this so that `set_unroll_depth` just takes a `(key, value)` pair (or whatever it actually needs)?

Reply:
On one hand, I guess a client could want to just run amok on this entry, on the other hand, maybe we want to constrain the client to just do what you're expected to do, which is decrement an unfold.

Not sure what is better.

Reply:
Are you referring to how we have

```
  let find_depth depth node = Env.Unroll_depth.find depth (Node.label node)
                              |> Option.value ~default:num_unroll
  in
  let decr_depth depth node =
    Env.Unroll_depth.update depth (Node.label node)
      ~f:(function None -> num_unroll - 1 | Some n -> n - 1)
  in
```
separately, and can make the api based on these rather than giving the client full reign over the map itself?

Reply:
Yes, though I'm not sure this is the right thing to do.

What's your feeling?